cloudinary/__init__.py

```diff
@@ -173,6 +133,65 @@ def _is_url_scheme_valid(url):
             return False
         return True
 
+    @staticmethod
+    def _parse_cloudinary_url(cloudinary_url):
+        return urlparse(cloudinary_url)
+
+    @abc.abstractmethod
+    def _config_from_parsed_url(self, parsed_url):
+        """Extract additional config from the parsed URL."""
+        raise NotImplementedError()
+
+    def _setup_from_parsed_url(self, parsed_url):
+        config = self._config_from_parsed_url(parsed_url)
```

Comment:
My IDE(Pycharm) tells me: `Shadows name 'config' from outer scope`, which is true.

Please fix this and a few other similar issues.